pages/checkout/index.js

```diff
@@ -117,32 +179,40 @@ export default function Checkout() {
       <Grid item xs={12} className={classes.gridItem}>
         <p>Total: {total}</p>
       </Grid>
+      {processingOrder ? (
+        <div>
+          <img src={'/spinner.gif'} style={{ width: '200px', margin: 'auto', display: 'block' }} alt="Processing Order..." />
+        </div>
+      ) : (
+        <div>
+          <Grid item xs={12} className={classes.gridItem}>
+            <Typography variant="h4" component="h4">
+              Shipping Address
+            </Typography>
+            <TextField required id="name" label="First Name" variant="standard" className={classes.textField} onChange={handleFormChange('firstName')} />
```

Comment:
Make sure to pass the `value={firstName}` prop to each of these form inputs as well. The way you're currently handling state is not how it should be done.